opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -620,6 +624,12 @@ namespace Opm {
                     R_sum[ contiPolymerEqIdx ] += R2;
                     maxCoeff[ contiPolymerEqIdx ] = std::max( maxCoeff[ contiPolymerEqIdx ], std::abs( R2 ) / pvValue );
                 }
+                if (has_foam_ ) {
+                    B_avg[ contiFoamEqIdx ] += 1.0 / fs.invB(FluidSystem::gasPhaseIdx).value();
+                    const auto R2 = ebosResid[cell_idx][contiFoamEqIdx];
```

Comment:
Does using B_avg for gas make sense? I guess the same questions applies to polymer and water. Anyway it probably doesn't matter much.

Reply:
I agree it is a bit shaky, but not likely to matter. I think it is ok since gas is the transport phase for the foam for now.